### ISSUE TITLE
[feature request] BatchNorm frozen mode in core

### ISSUE THREAD

**Issue by vadimkantorov** [pytorch/pytorch#66073]
This is a very frequent mode that's needed for finetuning as evidenced by classes in torchvision and detectron2: https://detectron2.readthedocs.io/en/latest/_modules/detectron2/layers/batch_norm.html#FrozenBatchNorm2d. It would be nice to have it in some unified version in core to avoid code duplication and have more visibility for this nuance to the user, and also have this functionality not only for 2d, but for 1d and 3d as well. This could be a method `freeze()` on the base class `_BatchNorm` and a variable `is_frozen` or `frozen` on the module instance

**Comment by ngimel** [comment]
You can put bn modules to `.eval()` mode to get this behavior (you still can backward through them).

**Comment by vadimkantorov** [comment]
The problem is that they can still be put in train() by further calls on parent modules. I've had some workarounds in my code about this, but it's so widely used that I think it makes sense to solve this problem in core

**Comment by vadimkantorov** [comment]
It may be easier to restructure the "modes" to make it more composable:

1) are running stats used in train? are running stats used in eval?
2) are running stats updated in train? (for symmetry: are running stats updated in eval?)

**Comment by vadimkantorov** [comment]
Also, the frozen mode in core could eliminate slight discrepancies between https://github.com/pytorch/vision/blob/d24ef4c40e31a233525b1b6a1082e8796688ae4c/torchvision/ops/misc.py#L89 and https://github.com/facebookresearch/detectron2/blob/cbbc1ce26473cb2a5cc8f58e8ada9ae14cb41052/detectron2/layers/batch_norm.py#L57

**Comment by vadimkantorov** [comment]
In addition, both torchvision's FrozenBatchNorm2d and detectron2's FrozenBatchNorm2d do not inherit from BatchNorm2d and are not susceptible to fusion as long as fusers check for base class.
